dev/release/verify-release-candidate.sh

```diff
@@ -499,6 +499,36 @@ maybe_setup_conda() {
   fi
 }
 
+install_maven() {
+  MAVEN_VERSION=3.8.7
+  SYSTEM_MAVEN_VERSION=$(mvn -v | head -n 1 | awk '{print $3}')
+
+  if [[ "$MAVEN_VERSION" == "$SYSTEM_MAVEN_VERSION" ]]; then
+      echo "Skip Install - versions are the same"
+  else
+      # the following logic only verifies released versions
+      older_version=$(printf '%s\n%s\n' "$SYSTEM_MAVEN_VERSION" "$MAVEN_VERSION" | sort --version-sort | head -n1)
```

Comment:
This solves the problem of properly sorting unreleased versions, as described [here](https://unix.stackexchange.com/a/687267):
```suggestion
      older_version=$(printf '%s\n%s\n' "$SYSTEM_MAVEN_VERSION" "$MAVEN_VERSION" | sed 's/$/|/' | sort -V | sed 's/|$//' | head -n1)
```

Reply:
Hmm. Do we need to add support for unreleased versions?
(I think that we don't need it for easy to maintain with simpler implementation.)

If we want to support unreleased versions by this approach, could you add a comment why we need to append `|` only while sorting?
(I don't want to use `|` here because it's one of special characters in extended regular expressions.)

Reply:
I think this is worthwhile. Without this, if someone is doing release verification and they have a new pre-release version of Maven installed, this could end up installing an older version of Maven.

I will add a comment explaining why this works.